control/controlclient/direct.go

```diff
@@ -1147,8 +1148,19 @@ func ipForwardingBroken(routes []netaddr.IPPrefix) bool {
 		return false
 	}
 
+	localIPs := map[netaddr.IP]bool{}
+	interfaces.ForeachInterfaceAddress(func(_ interfaces.Interface, pfx netaddr.IPPrefix) {
```

Comment:
This is pretty expensive compared to using the link monitor's already-cached version from the last update.

Even if you wanted to do it this way instead of passing the globally-shared-but-not-a-global link monitor in to this function, it'd be better to do this lazily upon the first time you hit a probably-pretty-rare IsSingleIP route.

Otherwise this slows down all map requests (even the "lite" non-streaming ones) unnecessarily.

Reply:
Done, using the link monitor you plumbed in another change.